frontend/appflowy_flutter/lib/workspace/application/settings/shortcuts/settings_shortcuts_cubit.dart

```diff
@@ -0,0 +1,72 @@
+import 'package:appflowy/workspace/application/settings/shortcuts/settings_shortcuts_service.dart';
+import 'package:appflowy_editor/appflowy_editor.dart';
+import 'package:collection/collection.dart';
+import 'package:equatable/equatable.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+enum ShortcutsStatus { initial, updating, success, failure }
+
+class ShortcutsCubit extends Cubit<ShortcutsState> {
+  ShortcutsCubit(this.service) : super(const ShortcutsState());
+
+  final SettingsShortcutService service;
+
+  Future<void> fetchShortcuts() async {
+    emit(state.copyWith(status: ShortcutsStatus.updating));
+    try {
+      final newCommandShortcuts = await service.loadShortcuts();
+      emit(
+        state.copyWith(
+          status: ShortcutsStatus.success,
+          commandShortcutEvents: newCommandShortcuts,
+        ),
+      );
+    } catch (e) {
+      //could also show an error
+      debugPrint("could not load ${e.toString()}");
+      emit(state.copyWith(status: ShortcutsStatus.failure));
+    }
+  }
+
+  Future<void> updateAllShortcuts() async {
+    emit(state.copyWith(status: ShortcutsStatus.updating));
+
+    try {
+      service.saveAllShortcuts(state.commandShortcutEvents);
+      emit(state.copyWith(status: ShortcutsStatus.success));
+    } catch (_) {
+      emit(state.copyWith(status: ShortcutsStatus.failure));
+    }
+  }
+
+  String getConflict(String command) {
+    final conflict = state.commandShortcutEvents
+        .firstWhereOrNull((el) => el.command == command);
+    return conflict != null ? conflict.key : '';
+  }
+}
+
+class ShortcutsState extends Equatable {
```

Comment:
As Alex mentioned, you don't need to manually implement the state if you use freezed to generate them.

Reply:
Done